actors/miner/src/lib.rs

```diff
@@ -4795,6 +4848,31 @@ fn verify_deals(
     ))?)
 }
 
+fn get_piece_manifest_from_deals(
+    rt: &impl Runtime,
+    deal_ids: &Vec<DealID>,
+) -> Result<Vec<(Cid, PaddedPieceSize)>, ActorError> {
+    let mut pieces: Vec<(Cid, PaddedPieceSize)> = vec![];
+
+    for deal_id in deal_ids {
+        let ret = get_deal_data_commitment(rt, deal_id)?;
+        pieces.push((ret.data, ret.size));
+    }
+    Ok(pieces)
+}
+
+fn get_deal_data_commitment(
+    rt: &impl Runtime,
+    deal_id: &DealID,
+) -> Result<ext::market::GetDealDataCommitmentReturn, ActorError> {
+    deserialize_block(extract_send_result(rt.send_simple(
```

Comment:
@anorth 

I think we should add a batched version of this method to the Market Actor so that the Miner Actor can lookup piece information for a array of deal_ids. This will reduce the number of calls we make to the Market Actor for emitting Sector Activation/Update events. Will have to change the FIP as well to add this new `GET_DEAL_DATA_COMMITMENT_BATCHED` method to the Market Actor ?